pages/blog/june-2019-update.md

```diff
@@ -0,0 +1,43 @@
+---
+description: Here are the most significant changes in June.
+---
+
+# June 2019 Update
+
+**Olivier Tassinari**
+
+*July 8, 2019*
+
+Here are the most significant changes in June:
+
+- 💄 We have introduced a new [ButtonGroup](/components/buttons/#grouped-buttons) component.
+  ![ButtonGroup](/static/blog/june-2019-update/button-group.png)
+
+- 💄 We have introduced a new [Slider](/components/slider/) component.
+  ![Slider](/static/blog/june-2019-update/slider.png)
+
+- 💄 We have introduced a new [TextareaAutosize](/components/textarea-autosize/) component.
+  <video src="/static/blog/june-2019-update/textarea-autosize.mp4" controls loop autoplay></video>
+
+But this summary is just scratching the surface. We have accepted 198 commits from 71 different contributors. We have changed 5,384 files with 26,199 additions and 18,097 deletions.
+
+## Our roadmap intent for July
+
+*(We'll do our best, no guarantee!)*
+
+- We will keep working on a new Tree View component.
+  You can already [preview it](https://deploy-preview-14827--material-ui.netlify.com/components/tree-view/).
+  ![Tree View](/static/blog/may-2019-update/tree-view.png)
+
+- We will keep working on a new Rating component.
+  You can already [preview it](https://deploy-preview-16455--material-ui.netlify.com/components/rating/).
+  ![Rating](/static/blog/june-2019-update/rating.png)
+
+- We will consider working on one of the following components:
+  - Autocomplete, Combo Box, Multi Select, Advanced Select
+  - Upload / Dropzone
+  - Carousel
```

Comment:
How were these decided? The latter two seem quite reactive (no pun intended)...

Reply:
They are coming from a analysis of the search patterns on Algolia June 25th, a discussion I had with Josh. Taking the carousel specifically, they are x5 more searches for it than for the rating (130 vs 25 over 1 week). To but in perspective with the 2000 searches for icons.

We can use this monthly pull request as a framework for defining the roadmap every month in the open. Nothing is decided until its merged, only a proposal at this stage.

I had the following top priorities in mind:
icon 1961 -> docs search for icons
grid 1907 -> improve impls, investigate css grid
upload 200 -> rich component
dropzone 22
carousel 134 -> sure
tree 66 -> 👌
rating 26 -> quick win?
naked
move system to core

Reply:
is there a way to make that info public or to post a screenshot? it would be neat to see the search volumes

Reply:
@rudolfolah The search analytics is available to members that have access to the Material-UI Algolia account. The data is raw, it also has to be deduplicated. Some components (Button, Grid & Icon) are leading the search volume. Also, there is a long tail of search. The most searched item accounts for 3% of all the searches.

Reply:
@mbrookes How do you want to proceed? Would you change the workflow or the topics of interest?

Reply:
I was unsure as to the rational, but since it's driven by search results, that seems good to me.

Reply:
Hello,
> icon 1961 -> docs search for icons
> grid 1907 -> improve impls, investigate CSS grid
> upload 200 -> rich component
> dropzone 22
> carousel 134 -> sure
> tree 66 -> 
> rating 26 -> quick win?
> naked
> move system to the core
> style adapters
> sidebar 150 -> better drawer?
> layout 73
> pagination 142 -> quick win?
> timeline 20 -> quick win?
> autocomplete 31
> combo box 13

I had to code several things from this list, here's a feedback:
- `dropzone` is really easy to do thanks to [react-dropzone](https://github.com/react-dropzone/react-dropzone)
- `upload` is a bit tricky to do with button, I had to google a lot and I was stumbling on v0.x answers on stackoverflow
- `icon`: everytime I want to find the icon list, I search icon in the docs to finally find the material.io link
- `tree`: I believe your ongoing effort will make it easier in the future
- `autocomplete/combo/multiselect`: by **far** the most annoying thing to do with material-ui, [react-select](https://github.com/JedWatson/react-select) is the most used solution but the package is not that easy to customize and full of bugs (see the issues and pr tabs)

Reply:
There's an upload button example in the docs...

Reply:
> sidebar 150 -> better drawer?

Or use the term sidebar as a synonym on the drawer page to make it discoverable for that search term?

Reply:
> There's an upload button example in the docs

Correct, I was referring to a richer upload component. For instance: https://www.telerik.com/kendo-react-ui/components/upload/file-processing/.

> Or use the term sidebar as a synonym on the drawer page to make it discoverable for that search term?

Oh yes, great idea.

Reply:
I was replying to @raphaelbadia, hence the quote; but yes, we can do more

Reply:
Oops, sorry.

Reply:
> pagination 142 -> quick win?

We could make the TablePagination usable standalone for an even quicker win

> timeline 20 -> quick win?

A timeline has a lot in common with the vertical stepper. I wonder if it would be worth sharing code?

Reply:
Maybe we should remove the last item (the components we might work on) so we. An leave it more time to mature?